src/query/sql/src/planner/optimizer/cascades/cascade.rs

```diff
@@ -51,9 +51,12 @@ impl CascadesOptimizer {
     pub fn create(
         ctx: Arc<dyn TableContext>,
         metadata: MetadataRef,
-        optimized: bool,
+        mut optimized: bool,
     ) -> Result<Self> {
         let explore_rule_set = if ctx.get_settings().get_enable_cbo()? {
+            if ctx.get_settings().get_disable_join_reorder()? {
+                optimized = true;
```

Comment:
When `optimized` is true, we will still use `rule_set_rs_l1` to cause join reorder.